scala/src/main/scala/smile/clustering/distance/BinaryDistance.scala

```diff
@@ -0,0 +1,12 @@
+package smile.math.distance
+
+/**
+ * @author Beck Gaël
+ * Trait which defines binary distance
+ * Here you can find a survey on binary distance that inherit from BinaryDistance : http://www.iiisci.org/journal/CV$/sci/pdfs/GS315JG.pdf
+ **/
+trait BinaryDistance extends Distance {
```

Comment:
This class is not necessary. And what about a distance is both binary and metric? This design creates difficulty for that.